### ISSUE TITLE
Allow retrofit of non-thermal resources

### ISSUE THREAD

**Issue by gschivley** [GenXProject/GenX#635]
In the new retrofit feature of #600, do not limit retrofits to thermal resources. This would be useful for other technologies, such as repowering wind or upgrading the generation capacity at hydroelectric dams.

**Comment by cfe316** [comment]
How would repowering wind be different from allowing capacity additions? Should I imagine it like
* Taller wind farms --> access to better variability-curves, for example?
* Adding capacity for the new curves necessitates removing capacity of the old shorter turbines.

**Comment by gschivley** [comment]
Yes, something like that. We don't allow new-build where wind farms already exist, which are generally going to be good locations. So it's possible that as existing wind starts to retire we can use higher hub heights to get more consistent generation profiles. 

I'm not sure if higher hub heights also allow for increased capacity. If capacity stays the same then interconnection infrastructure can be reused, reducing total costs. Something to look into.

**Comment by gschivley** [comment]
@lbonaldo and @sambuddhac can this issue be closed with the 0.4 release?

**Comment by lbonaldo** [comment]
Yes, the latest retrofit module introduced in version 0.4 enables the retrofit of non-thermal generators. So this issue can be closed. There is only a note when using retrofitting with multistage and setting `myopic = 0` that is described in the documentation at the end of this [section](https://genxproject.github.io/GenX.jl/dev/User_Guide/multi_stage_input/#Input-data-files).
